src/silly_sprites/opengl/window.cpp

```diff
@@ -14,44 +14,39 @@ namespace sly::gl {
         }
     } // namespace
 
-    Window::Window(GlfwContext context, GLFWwindow* window)
-        : m_context{ std::move(context) },
-          m_window{ window, destroy_glfw_window } { }
-
-    [[nodiscard]] GLFWwindow* Window::get() const {
-        return m_window.get();
-    }
-
-    [[nodiscard]] Window Window::create(int const width, int const height) {
-        auto context = GlfwContext::create();
-        if (not context.has_value()) {
-            spdlog::critical("Unable to create GLFW context");
-            throw GLError{ GLErrorType::UnableToCreateGlfwContext };
-        }
-
+    Window::Window(int const width, int const height) : m_window{ nullptr, destroy_glfw_window } {
+ 
         glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
         glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
         glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
 
         GLFWwindow* const window = glfwCreateWindow(width, height, "coder2k bester Mann", nullptr, nullptr);
         if (window == nullptr) {
             spdlog::critical("Failed to create GLFW window");
-            throw GLError{ GLErrorType::FailedToCreateWindow };
+            throw GlError{ GlErrorType::FailedToCreateWindow };
         }
+        m_window = { window, destroy_glfw_window };
         glfwMakeContextCurrent(window);
         glfwSetInputMode(window, GLFW_STICKY_KEYS, GLFW_TRUE);
         glfwSetInputMode(window, GLFW_STICKY_MOUSE_BUTTONS, GLFW_TRUE);
 
         if (not gladLoadGL(glfwGetProcAddress)) {
             spdlog::critical("Failed to initialize Glad");
-            throw GLError{ GLErrorType::FailedToInitializeGlad };
+            throw GlError{ GlErrorType::FailedToInitializeGlad };
         }
 
-        gl::on_framebuffer_size_changed(window, width, height);  // addressing the gl namespace becuase otherwise the function is ambiguous
-        glfwSetFramebufferSizeCallback(window, gl::on_framebuffer_size_changed);
+        on_framebuffer_size_changed(
+                window,
+                width,
+                height
+        );
+        glfwSetFramebufferSizeCallback(window, on_framebuffer_size_changed);
 
         spdlog::info("window initialized");
-        return Window{ *std::move(context), window };
+    }
```

Comment:
In such a case, I prefer to extract creation of the `unique_ptr` into a static member function, so that, inside the constructor, we don't have a two-step-initialization (first with `nullptr` and later with the *real* pointer).
I mean something like this:
```cpp
    // static member function:
    std::unique_ptr<GLFWwindow, Deleter>Window::create(int const widt, int const height) {
        glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
        glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
        glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);

        auto const window = glfwCreateWindow(width, height, "coder2k bester Mann", nullptr, nullptr);
        if (window == nullptr) {
            spdlog::critical("Failed to create GLFW window");
            throw GlError{ GlErrorType::FailedToCreateWindow };
        }
        glfwMakeContextCurrent(window);
        glfwSetInputMode(window, GLFW_STICKY_KEYS, GLFW_TRUE);
        glfwSetInputMode(window, GLFW_STICKY_MOUSE_BUTTONS, GLFW_TRUE);

        if (not gladLoadGL(glfwGetProcAddress)) {
            spdlog::critical("Failed to initialize Glad");
            throw GlError{ GlErrorType::FailedToInitializeGlad };
        }

        on_framebuffer_size_changed(
                window,
                width,
                height
        );
        glfwSetFramebufferSizeCallback(window, on_framebuffer_size_changed);

        spdlog::info("window initialized");
        return { window, destroy_glfw_window };
    }

    // constructor:
    Window::Window(int const width, int const height) : m_window{ Window::create(width, height) } { }
```